src/java/com/zimbra/cs/taglib/tag/LoginTag.java

```diff
@@ -202,6 +207,16 @@ public void doTag() throws JspException, IOException {
                         mRememberMe,
                         mbox.getAuthResult().getExpires());
             }
+            final ZGetInfoResult accountInfo = mbox.getAccountInfo(false);
+            String publicUrlBase = accountInfo.getPublicURLBase();
+            URL url = new URL(publicUrlBase);
+            if (!url.getHost().equalsIgnoreCase(request.getServerName())) {
+                publicUrlBase = request.getScheme().toLowerCase() + "://" + url.getHost();
```

Comment:
Do we need to worry about ports?  Probably not but worth asking the question.

Reply:
I can add them if they are different from the standard port for the scheme.
i.e. 443 - https 
80 - http

Anything other than those would be added otherwise left out.